src/map/mob.cpp

```diff
@@ -2231,14 +2290,9 @@ static int32 mob_ai_sub_lazy(struct mob_data *md, va_list args)
 		return 0;
 	}
 
-	if (md->ud.walktimer == INVALID_TIMER) {
-		// Because it is not unset when the mob finishes walking.
-		mob_setstate(*md, MSS_IDLE);
-		if (md->idle_event[0] && npc_event_do_id( md->idle_event, md->bl.id ) > 0) {
-			md->idle_event[0] = 0;
-			return 0;
-		}
-	}
+	// Remove target and set to idle state when movement was finished
+	// This also triggers the idle NPC event
+	mob_unlocktarget(md, tick);
```

Comment:
Unrelated change.

Reply:
It's related because we clear the target here to prevent it from attempting warp chase again.